test/jpypetest/test_array.py

```diff
@@ -601,3 +602,24 @@ def testShortcut(self):
     def testJArrayIndex(self):
         with self.assertRaises(TypeError):
             jpype.JArray[10]
+
+    def testJArrayGeneric(self):
+        self.assertEqual(type(JObject[1]), JArray[JObject])
+
+    def testJArrayGeneric_Init(self):
+        Arrays = JClass("java.util.Arrays")
+        self.assertTrue(Arrays.equals(JObject[0], JArray[JObject](0)))
+
+    def testJArrayInvalidGeneric(self):
+        with self.assertRaises(TypeError):
+            jpype.JArray[object]
+    
+    def testJArrayGenericJClass(self):
+        self.assertEqual(type(JClass[0]), JArray[JClass])
+
+    def testJArrayGenericString(self):
+        self.assertEqual(type(JClass[0]), JArray["java.lang.Class"])
+
+    def testJArrayGenericStringInvalid(self):
+        with self.assertRaises(TypeError):
+            JArray["foo.bar"] # type: ignore
```

Comment:
I'm surprised there needs to be a type ignore here, given it is fine elsewhere to pass a str?

Reply:
You will get a warning that `foo.bar` doesn't exist, which happens to be the point of the test.